src/Community.VisualStudio.Toolkit.Shared/ExtensionMethods/WindowExtensions.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Threading.Tasks;
+using System.Windows;
+using Microsoft;
+using Microsoft.Internal.VisualStudio.PlatformUI;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Community.VisualStudio.Toolkit.Shared.ExtensionMethods
+{
+    /// <summary>Extension methods for the <see cref="Window"/> class.</summary>
+    public static class WindowExtensions
+    {
+        /// <summary>Shows a window as a dialog.</summary>
+        public static async Task<bool?> ShowDialogAsync(this Window window, WindowStartupLocation windowStartupLocation = WindowStartupLocation.CenterOwner)
+        {
+            if (window == null)
+                throw new ArgumentNullException(nameof(window));
+
+            var vsUiShellService = await VS.GetServiceAsync<SVsUIShell, IVsUIShell>();
+            Assumes.Present(vsUiShellService);
+
+            if (vsUiShellService.GetDialogOwnerHwnd(out var hwnd) != VSConstants.S_OK)
```

Comment:
Using `ErrorHandler.ThrowOnFailure` would be a better choice here (and below), as it will get the error message based on the error code.

```cs
ErrorHandler.ThrowOnFailure(vsUiShellService.GetDialogOwnerHwnd(out var hwnd));
```

Reply:
That wouldn't be needed if it was based on WPF's `DialogWIndow` instead of the COM-based `VSUtilities.ShowMessage()`..

Reply:
@reduckted - Thanks for pointing this out.  Wasn't aware of the helper method.  Updated now.